### ISSUE TITLE
[Brainstorming] Custom Format

### ISSUE THREAD

**Issue by yorkxin** [yorkxin/copy-as-markdown#133]
Supporting custom format has been asked in multiple feature requests.

This post collects some ideas of template variables.

- Current Date / Time (used as Access Date) #87
- Hostname #121 
- Index (`1`, `2`, `3`...) #92 (though I think reference-style links can be a built-in feature) 
- Image base64 #83 
- Text Selection #64 

Issues:

- Need to provide some snippets so that people can grab common formats from Wiki
- Writing template string means it might not be friendly to non-programmer 
- Number of custom formats
- How do they appear on the popup window
- How do they appear on the context menu
- Max items of keyboard shortcuts

**Comment by eloycoto** [comment]
Hi,

Could we create a template for selection text? Like I would like to get the selection and get the link for that, like:

```
[${url}][${title}]
Quote: ${quote}
```

Having templates like this, will allow users to have a lot of customization, like adding dates or functions :)
Thanks

**Comment by yorkxin** [comment]
Hi @eloycoto , thanks for the idea. Quote text (selection text) definitely will be helpful!

**Comment by dylanmorroll** [comment]
I agree something for selection text would be nice, I would like to be able to add simple markdown quote syntax (`>`) to each line. If you could iterate over the lines in the selected text that would be pretty nice.
